torch/utils/cpp_extension.py

```diff
@@ -1310,13 +1422,22 @@ def load(name,
     work fine. If not, setting the ``CUDA_HOME`` environment variable is the
     safest option.
 
+    SYCL support with mixed compilation is provided. Simply pass SYCL source
+    files (``.sycl``) along with other sources. Such files will be detected
+    and compiled with icpx (Intel DPC++ sycl compiler) rather than the C++
+    compiler. You can pass additional flags to icpx via ``extra_sycl_cflags``,
+    just like with ``extra_cflags`` for C++. icpx compiler is expected to
+    be found via system PATH environment variable.
+
     Args:
         name: The name of the extension to build. This MUST be the same as the
             name of the pybind11 module!
         sources: A list of relative or absolute paths to C++ source files.
         extra_cflags: optional list of compiler flags to forward to the build.
         extra_cuda_cflags: optional list of compiler flags to forward to nvcc
             when building CUDA sources.
+        extra_sycl_cflags: optional list of compiler flags to forward to icpx
```

Comment:
`icpx` is a new term. We need to unify the sycl compiler term. Either `dpc++` or `icpx`. Meanwhile, I remember that the compiler name is `icx` on windows.

Reply:
I reworded from icpx to DPC++ compiler here and in few other places in comments.